File: app/src/main/java/com/codekk/mvp/presenter/PresenterManager.java
package com.codekk.mvp.presenter;

import android.support.annotation.MenuRes;
import android.support.annotation.NonNull;

/**
 * by y on 2017/5/23.
 */

public interface PresenterManager {

    interface BlogListPresenter {
        void netWorkRequest(int page);
    }

    interface JobListPresenter {
        void netWorkRequest(int page);
    }

    interface MainPresenter {
        void switchId(@MenuRes int menuRes);

        void onBackPressed();

        void onMainDestroy();
    }

    interface OpaListPresenter {
        void netWorkRequest(int page);
    }

    interface OpaSearchPresenter {
        void netWorkRequest(@NonNull String text, int page);
    }

    interface OpDetailPresenter {
        void netWorkRequest(@NonNull String id, int type);
    }

    interface OpListPresenter {
        void netWorkRequest(int page);
    }

    interface OpSearchPresenter {
        void netWorkRequest(@NonNull String text, int page);
    }

    interface RecommendListPresenter {
        void netWorkRequest(int page);
    }

    interface RecommendSearchPresenter {
        void netWorkRequest(@NonNull String name, int page);
    }

}

File: app/src/main/java/com/codekk/mvp/presenter/PresenterManager.kt
package com.codekk.mvp.presenter

interface BlogListPresenter {
    fun netWorkRequest(page: Int)
}

interface JobListPresenter {
    fun netWorkRequest(page: Int)
}

interface OpaListPresenter {
    fun netWorkRequest(page: Int)
}

interface OpaSearchPresenter {
    fun netWorkRequest(text: String, page: Int)
}

interface OpDetailPresenter {
    fun netWorkRequest(id: String, type: Int)
}

interface OpListPresenter {
    fun netWorkRequest(page: Int)
}

interface OpSearchPresenter {
    fun netWorkRequest(text: String, page: Int)
}

interface RecommendListPresenter {
    fun netWorkRequest(page: Int)
}

interface RecommendSearchPresenter {
    fun netWorkRequest(name: String, page: Int)
}
